cpmpy/transformations/linearize.py

```diff
@@ -257,6 +326,25 @@ def linearize_constraint(lst_of_expr, supported={"sum","wsum"}, reified=False):
 
     return newlist
 
+def _linearize_abs(expr):
+    """
+        Linearize abs(lhs) == rhs
+        Takes as input an absolute value comparison, and returns a list of equivalent constraints
+    """
+    assert isinstance(expr, Comparison)
+    lhs, rhs = expr.args
+    assert isinstance(lhs, GlobalFunction) and lhs.name == "abs", f"Expected abs expr but got {lhs}"
+
+    lb, ub = get_bounds(lhs)
+    if lb >= 0: # always positive
+        return [lhs == rhs]
+    if ub <= 0: # always negative
+        return [lhs == -rhs]
+
+    lhs_is_pos = cp.boolvar()
+    return [lhs_is_pos.implies(lhs >= 0), (~lhs_is_pos).implies(lhs < 0),
```

Comment:
lhs < 0 is not linear right? I think lhs <= 0 or lhs <= -1 are both correct here since 0 = -0 anyway.

Reply:
Hmm you're right indeed `<` is not linear, it should not really matter here as we pass it through linearize either way. But when we allow it to be `<= 0` the `lhs_is_pos` is not fully defined...
I'll set it to `lhs <= -1`.